src/projection.jl

```diff
@@ -241,6 +241,12 @@ function (project::ProjectTo{AbstractArray})(dx::Number) # ... so we restore fro
     return fill(project.element(dx))
 end
 
+# Accept the Tangent corresponding to a Tuple -- Zygote's splats produce these
+function (project::ProjectTo{AbstractArray})(dx::Tangent{<:Any, <:Tuple})
```

Comment:
I think i prefer this to be written as:
```suggestion
function (project::ProjectTo{AbstractArray})(dx::Tangent{<:Tuple})
```
so we don't need to mention how this backed.

Reply:
I think Zygote managed to produce a `Tangent{Any, Tuple{...}}`, which was why my pirate method specified that, here:

https://github.com/FluxML/Zygote.jl/pull/1044/files#diff-e0bc7da8f1a33a59f5ecfa67257c04038f0b4915b3f74bdf39780818fd0010a2R162

But I was squashing bugs as fast as I could & didn't track it down any further. Can circle back now that things basically pass.

Reply:
right, fair enough. 
If we have to do it this way that is also fine.
Maybe just a commond is woth adding to clarify that this is the tangent for something with primal type `tangent`

Zygote does lose primal type information, that is a thing
We might well be able to teach it a bit more about tuples more easily than the general case.

Reply:
BTW, easy examples of this Tangent{Any} problem:
```
julia> using Zygote

julia> function Zygote.ChainRulesCore.rrule(::typeof(identity), x::AbstractArray)
         x, dx -> (Zygote.NoTangent(), @show dx)
       end

julia> Zygote.gradient(x -> max(identity(x)...), [1,2,3])
dx = Tangent{Any}(0, 0, 1)
((0, 0, 1),)

julia> Zygote.gradient(x -> sum(identity(x).parent), [1,2,3]')
dx = Tangent{Any}(parent = 3-element Fill{Int64}: entries equal to 1,)
((parent = 3-element Fill{Int64}: entries equal to 1,),)
```